src/pages/index.js

```diff
@@ -1,248 +1,250 @@
-});
+let cardSection;
 
-const cardSection = new Section({ items: cardData }, renderCard, "#section");
-cardSection.renderItems(cardData);
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection = new Section({ items: cards }, renderCard, "#section");
 
-const newUserInfo = new UserInfo("#profilename", "#profilejob");
+    function renderCard(cards) {
+      const cardElement = createCard(cards);
+      cardSection.addItem(cardElement);
+    }
 
-function renderCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleImageClick);
-  // cardListEl.prepend(cardElement.getView());
-  cardSection.addItem(cardElement.getView());
-}
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
 
-function handleImageClick(cardData) {
-  console.log(cardData);
-  contentCardPreview.openModal(cardData);
-}
+const newCardPopup = new PopupWithForm({
+  popupSelector: "#contentModal",
+  handleFormSubmit: handleCardFormSubmit,
+});
+
+contentAddButton.addEventListener("click", () => {
+  cardFormValidator.toggleButtonState();
+  newCardPopup.openModal();
+});
 
-// ---- Profile Modal open and close event listeners -------------------------------------
+function handleCardFormSubmit(cardValues) {
+  const submitButton = document.getElementById("modalContentSubmit");
+  const originalButtonText = submitButton.textContent;
+  // Change button text to "Saving..."
+  submitButton.textContent = "Saving...";
+
+  api
+    .addNewCard(cardValues)
+    .then((cardData) => {
+      const newCard = createCard(cardData);
+      cardSection.prependItems(newCard);
+      newCardPopup.closeModal();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitButton.textContent = originalButtonText;
+    });
+}
 
-const editProfileButton = document.querySelector(
-  "#profile__edit-button-action"
+const newUserInfo = new UserInfo(
+  "#profilename",
+  "#profilejob",
+  "#profileImage"
 );
 
+api
+  .loadUserInfo()
+  .then((userData) => {
+    newUserInfo.setUserInfo({
+      name: userData.name,
+      about: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
 editProfileButton.addEventListener("click", () => {
-  // nameInput.value = profileName.textContent;
-  // jobInput.value = profileJob.textContent;
   const data = newUserInfo.getUserInfo();
   nameInput.value = data.name;
   jobInput.value = data.job;
-  cardFormValidator.toggleButtonState();
   profileCardPopup.openModal();
 });
 
-// profileEditModalClose.addEventListener("click", () =>
-//   profileCardPopup.closeModal()
-// );
+profilePhoto.addEventListener("click", () => {
+  profilePhotoEditValidator.toggleButtonState();
+  profilePhotoEdit.openModal();
+});
 
-// modalPreviewCloseButton.addEventListener("click", () =>
-//   contentCardPreview.closeModal()
-// );
+const profileCardPopup = new PopupWithForm({
```

Comment:
`profileCardPopup`  should be named just `profilePopup` (it has nothing to do with the cards). It's misleading